src/tests/java/htsjdk/samtools/sra/SRATest.java

```diff
@@ -0,0 +1,551 @@
+
+            // primary and secondary alignments
+            {"SRR833251", 81, 393, "SRR833251.R.51",
+                    "ATGCAAATCCGAATGGGCTATTTGTGGGTACTTGGGCAGGTAAGTAGCTGGCAATCTTGGTCGGTAAACCAATACCCAAGTTCACATAGGCACCATCGGGA",
+                    "CCCFFFFFHHHHHIJJJIJJJJJIIJJJGIJIJIIJIJJJDGIGIIJIJIHIJJJJJJGIGHIHEDFFFFDDEEEDDDDDCDEEDDDDDDDDDDDDDBBDB",
+                    1787186, "38M63S", "gi|169794206|ref|NC_010410.1|", 11, true, true},
+        };
+    }
+
+    @Test(dataProvider = "testRows")
+    public void testRows(String acc, int recordIndex, int flags, String readName, String bases, String quals, int refStart, String cigar,
+                         String refName, int mapQ, boolean hasMate, boolean isSecondaryAlignment) {
+        if (!SRAAccession.isSupported()) return;
+
+        SamReader reader = SamReaderFactory.make().validationStringency(ValidationStringency.SILENT).open(
+                SamInputResource.of(new SRAAccession(acc))
+        );
+
+        final SAMRecordIterator samRecordIterator = reader.iterator();
+
+        while (recordIndex != 0) {
+            Assert.assertTrue(samRecordIterator.hasNext(), "Record set is too small");
+
+            samRecordIterator.next();
+            recordIndex--;
+        }
+        Assert.assertTrue(samRecordIterator.hasNext(), "Record set is too small");
+
+        SAMRecord record = samRecordIterator.next();
+
+        List<SAMValidationError> validationErrors = record.isValid();
+        Assert.assertNull(validationErrors, "SRA Lazy record is invalid. List of errors: " + (validationErrors != null ? validationErrors.toString() : ""));
+
+        Assert.assertEquals(record.getReadName(), readName);
+        Assert.assertEquals(new String(record.getReadBases()), bases);
+        Assert.assertEquals(record.getBaseQualityString(), quals);
+        Assert.assertEquals(record.getReadPairedFlag(), hasMate);
+        Assert.assertEquals(record.getFlags(), flags);
+        Assert.assertEquals(record.getNotPrimaryAlignmentFlag(), isSecondaryAlignment);
+        if (refStart == -1) {
+            Assert.assertEquals(record.getReadUnmappedFlag(), true);
+            Assert.assertEquals(record.getAlignmentStart(), 0);
+            Assert.assertEquals(record.getCigarString(), "*");
+            Assert.assertEquals(record.getReferenceName(), "*");
+            Assert.assertEquals(record.getMappingQuality(), 0);
+        } else {
+            Assert.assertEquals(record.getReadUnmappedFlag(), false);
+            Assert.assertEquals(record.getAlignmentStart(), refStart);
+            Assert.assertEquals(record.getCigarString(), cigar);
+            Assert.assertEquals(record.getReferenceName(), refName);
+            Assert.assertEquals(record.getMappingQuality(), mapQ);
+        }
+    }
+
+    @Test(dataProvider = "testRows")
+    public void testRowsAfterIteratorDetach(String acc, int recordIndex, int flags, String readName, String bases, String quals,
```

Comment:
The code in this test is identical to the one above except for one line -- can you have both tests call into a shared method for most of the work?